hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/replication/GrpcOutputStream.java

```diff
@@ -147,10 +149,12 @@ private void flushBuffer(boolean eof) {
    * the stream until it's ready.
    */
   private void waitUntilReady() {
-    while (!streamObserver.isReady()) {
+    int count = 0;
+    while (!streamObserver.isReady() && count < READY_RETRY_COUNT) {
```

Comment:
Won't this lead to the same memory bloat? Retrying the whole container replication might be better if the channel is busy for a while.